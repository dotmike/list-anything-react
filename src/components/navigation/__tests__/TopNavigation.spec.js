import configureStore from "redux-mock-store";

import ConnectedTopNavigation, {
  UnconnectedTopNavigation
} from "../TopNavigation";

const setup = (propOverrides = {}) => {
  const props = {
    toggleMenu: jest.fn(),
    hideMenu: jest.fn(),
    search: jest.fn(),
    activeItem: "",
    email: "main@mail.com",
    hasSubjects: false,
    logout: jest.fn(),
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(
      ConnectedTopNavigation,
      props,
      true
    ),
    wrapperShallow: wrapperShallow(UnconnectedTopNavigation, props, true)
  };
};

describe("ConnectedTopNavigation", () => {
  const mockStore = configureStore();
  const state = {
    user: {
      email: "main@mail.com"
    }
  };
  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedTopNavigation", () => {
  it("renders correctly when `hasSubjects` is false", () => {
    const { wrapperShallow: wrapper } = setup({ hasSubjects: false });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `hasSubjects` is true", () => {
    const { wrapperShallow: wrapper } = setup({ hasSubjects: true });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `activeItem` is set", () => {
    const { wrapperShallow: wrapper } = setup({ activeItem: "/dashboard" });
    expect(wrapper()).toMatchSnapshot();
  });

  it("calls `toggleMenu` when clicking on the `Sidemenu MenuItem`", () => {
    const { wrapperShallow: wrapper, props } = setup();

    wrapper()
      .find("MenuItem")
      .find({ name: "toggle-sidebar" })
      .simulate("click");
    expect(props.toggleMenu).toHaveBeenCalled();
  });

  it("calls `hideMenu` when clicking on a `MenuItem`", () => {
    const { wrapperShallow: wrapper, props } = setup();

    wrapper()
      .find("MenuItem")
      .find({ to: "/dashboard" })
      .simulate("click");
    expect(props.hideMenu).toHaveBeenCalled();
  });

  it("calls `logout` when clicking on the `logout DropdownItem`", () => {
    const { wrapperShallow: wrapper, props } = setup();

    wrapper()
      .find("DropdownItem")
      .find({ name: "logout-btn" })
      .simulate("click");
    expect(props.logout).toHaveBeenCalled();
  });
});
