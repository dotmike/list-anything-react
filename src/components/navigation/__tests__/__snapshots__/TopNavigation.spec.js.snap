// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedTopNavigation renders correctly 1`] = `
<TopNavigation
  activeItem=""
  email="main@mail.com"
  hasSubjects={false}
  hideMenu={[Function]}
  logout={[Function]}
  search={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  toggleMenu={[Function]}
/>
`;

exports[`UnconnectedTopNavigation renders correctly when \`activeItem\` is set 1`] = `
<Menu
  attached="top"
  id="top-navigation"
  inverted={true}
  size="small"
>
  <MenuItem
    onClick={[Function]}
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="sidebar"
    />
    <span
      className="hide-lt-767px"
    >
      Menu
    </span>
  </MenuItem>
  <MenuItem
    active={true}
    as={[Function]}
    onClick={[Function]}
    to="/dashboard"
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="home"
    />
    <span
      className="hide-lt-767px"
    >
      Dashboard
    </span>
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    onClick={[Function]}
    to="/subjects/new"
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="plus"
    />
    <span
      className="hide-lt-767px"
    >
      Add new Subject
    </span>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <MenuItem
      className="hide-lt-767px"
    >
      <SearchInput
        id="seach-form-topnav"
        onSearch={[Function]}
      />
    </MenuItem>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      icon="dropdown"
      item={true}
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      trigger={
        <span>
          <Image
            as="img"
            avatar={true}
            src="https://gravatar.com/avatar/ad3ddf5072a6de3dcffc3b2c7c2cec04?size=30"
            ui={true}
          />
        </span>
      }
    >
      <DropdownMenu>
        <DropdownItem
          onClick={[Function]}
        >
          Logout
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </MenuMenu>
</Menu>
`;

exports[`UnconnectedTopNavigation renders correctly when \`hasSubjects\` is false 1`] = `
<Menu
  attached="top"
  id="top-navigation"
  inverted={true}
  size="small"
>
  <MenuItem
    onClick={[Function]}
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="sidebar"
    />
    <span
      className="hide-lt-767px"
    >
      Menu
    </span>
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    onClick={[Function]}
    to="/dashboard"
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="home"
    />
    <span
      className="hide-lt-767px"
    >
      Dashboard
    </span>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <MenuItem
      className="hide-lt-767px"
    >
      <SearchInput
        id="seach-form-topnav"
        onSearch={[Function]}
      />
    </MenuItem>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      icon="dropdown"
      item={true}
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      trigger={
        <span>
          <Image
            as="img"
            avatar={true}
            src="https://gravatar.com/avatar/ad3ddf5072a6de3dcffc3b2c7c2cec04?size=30"
            ui={true}
          />
        </span>
      }
    >
      <DropdownMenu>
        <DropdownItem
          onClick={[Function]}
        >
          Logout
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </MenuMenu>
</Menu>
`;

exports[`UnconnectedTopNavigation renders correctly when \`hasSubjects\` is true 1`] = `
<Menu
  attached="top"
  id="top-navigation"
  inverted={true}
  size="small"
>
  <MenuItem
    onClick={[Function]}
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="sidebar"
    />
    <span
      className="hide-lt-767px"
    >
      Menu
    </span>
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    onClick={[Function]}
    to="/dashboard"
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="home"
    />
    <span
      className="hide-lt-767px"
    >
      Dashboard
    </span>
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    onClick={[Function]}
    to="/subjects/new"
  >
    <Icon
      as="i"
      className="menu-icon-responsive"
      name="plus"
    />
    <span
      className="hide-lt-767px"
    >
      Add new Subject
    </span>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <MenuItem
      className="hide-lt-767px"
    >
      <SearchInput
        id="seach-form-topnav"
        onSearch={[Function]}
      />
    </MenuItem>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      icon="dropdown"
      item={true}
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      trigger={
        <span>
          <Image
            as="img"
            avatar={true}
            src="https://gravatar.com/avatar/ad3ddf5072a6de3dcffc3b2c7c2cec04?size=30"
            ui={true}
          />
        </span>
      }
    >
      <DropdownMenu>
        <DropdownItem
          onClick={[Function]}
        >
          Logout
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </MenuMenu>
</Menu>
`;
